tools/objc/libtool.sh

```diff
@@ -67,62 +68,77 @@ if [[ ! -x "$python_executable" ]]; then
-    # pass a new filelist which contains the symlinks.
-    -filelist)
-      ARGS+=("${ARG}")
-      ARG="$1"
-      shift
-      HASHED_FILELIST="${ARG%.objlist}_hashes.objlist"
-      rm -f "${HASHED_FILELIST}"
-      # Use python helper script for fast md5 calculation of many strings.
-      "$python_executable" "${MY_LOCATION}/make_hashed_objlist.py" \
-        "${ARG}" "${HASHED_FILELIST}" "${TEMPDIR}"
-      ARGS+=("${HASHED_FILELIST}")
-      ;;
-   # Output flag
-    -o)
-     ARGS+=("${ARG}")
-     ARG="$1"
-     shift
-     ARGS+=("${ARG}")
-     OUTPUTFILE="${ARG}"
-     ;;
-   # Flags with no args
-    -static|-s|-a|-c|-L|-T|-D|-no_warning_for_no_symbols)
-      ARGS+=("${ARG}")
+function parse_option() {
+  local -r ARG="$1"
+  if [[ "$handle_filelist" == "1" ]]; then
+    handle_filelist=0
+    HASHED_FILELIST="${ARG%.objlist}_hashes.objlist"
+    rm -f "${HASHED_FILELIST}"
+    # Use python helper script for fast md5 calculation of many strings.
+    "$python_executable" "${MY_LOCATION}/make_hashed_objlist.py" \
+      "${ARG}" "${HASHED_FILELIST}" "${TEMPDIR}"
+    ARGS+=("${HASHED_FILELIST}")
+  elif [[ "$keep_next" == "1" ]]; then
+    keep_next=0
+    ARGS+=("$ARG")
+  else
+    case "${ARG}" in
+      # Filelist flag, need to symlink each input in the contents of file and
+      # pass a new filelist which contains the symlinks.
+      -filelist)
+        handle_filelist=1
+        ARGS+=("${ARG}")
+        ;;
+      @*)
+        use_params_file=1
+        path="${ARG:1}"
+        while IFS= read -r opt
+        do
+          parse_option "$opt"
+        done < "$path" || exit 1
+        ;;
+      # Output flag
+      -o)
+        keep_next=1
+        ARGS+=("${ARG}")
       ;;
-   # Single-arg flags
-   -arch_only|-syslibroot)
-     ARGS+=("${ARG}")
-     ARG="$1"
-     shift
-     ARGS+=("${ARG}")
-     ;;
-   # Any remaining flags are unexpected and may ruin flag parsing.
-   -*)
-     echo "Unrecognized libtool flag ${ARG}"
-     exit 1
-     ;;
-   # Archive inputs can remain untouched, as they come from other targets.
-   *.a)
-     ARGS+=("${ARG}")
-     ;;
-   # Remaining args are input objects
-   *)
-     ARGS+=("$(echo "$(hash_objfile "${ARG}")")")
-     ;;
-   esac
+      # Flags with no args
+      -static|-s|-a|-c|-L|-T|-D|-no_warning_for_no_symbols)
+        ARGS+=("${ARG}")
+        ;;
+      # Single-arg flags
+      -arch_only|-syslibroot)
+        keep_next=1
+        ARGS+=("${ARG}")
+        ;;
+      # Any remaining flags are unexpected and may ruin flag parsing.
+      -*)
+        echo "Unrecognized libtool flag ${ARG}"
+        exit 1
+        ;;
+      # Archive inputs can remain untouched, as they come from other targets.
+      *.a)
+        ARGS+=("${ARG}")
+        ;;
+      # Remaining args are input objects
+      *)
+        ARGS+=("$(hash_objfile "${ARG}")")
+        ;;
+    esac
+  fi
+}
+
+for arg in "$@"; do
+  parse_option "$arg"
 done
 
+if [[ "$use_params_file" == "1" ]]; then
```

Comment:
maybe we should just always do this at this point?

Reply:
that seems reasonable